bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -295,7 +455,64 @@ public Point closest(Point p) {
         }
     }
 
+    private static boolean inByteRange(int val) {
+        return val >= 0 && val <= 255;
+    }
+
     private static boolean inRange(double val) {
         return val >= 0.0 && val <= 1.0;
     }
+
+    private static int convertColorPercentToByte(PercentType percent) {
+        return percent.toBigDecimal().multiply(BigDecimal.valueOf(255))
+                .divide(BIG_DECIMAL_HUNDRED, 0, RoundingMode.HALF_UP).intValue();
+    }
+
+    /* Utility functions for comparing color values */
+
+    /**
+     * Helper method for checking if two HSBType colors are close to each other. A maximum deviation is specifid in
+     * percent.
+     *
+     * @param expected an HSBType containing the first color.
+     * @param actual an HSBType containing the second color.
+     * @param delta the maximum allowed percentage difference between the two (1..99 percent).
+     * @throws IllegalArgumentException if delta is out of range.
+     */
+    public static boolean closeToHsb(HSBType expected, HSBType actual, float delta) throws IllegalArgumentException {
+        if (delta <= 0f || delta > 99f) {
+            throw new IllegalArgumentException("'delta' out of bounds");
+        }
+        double[] exp = ColorUtil.hsbToXY(expected);
+        double[] act = ColorUtil.hsbToXY(actual);
+        double max = delta / 100.0f;
+        return ((Math.abs(exp[0] - act[0]) < max) && (Math.abs(exp[1] - act[1]) < max));
+    }
+
+    /**
+     * Helper method for checking if two RGB colors (int[3], 0..255) are close to each other. A maximum distance is
+     * specified
+     * as sum of squared distances of the integer components.
+     *
+     * @param first an int array containing the first RGB color.
+     * @param second an int array containing the second RGB color.
+     * @param maxSquaredSum the maximum allowed squared sum of differences.
+     * @throws IllegalArgumentException if array size is not 3.
+     */
+    public static boolean closeToRgb(final int[] firstRgb, final int[] secondRgb, int maxSquaredSum)
```

Comment:
This should be removed. Bindings should neither test color conversion (as that is part of core), nor should they handle RGB in any other way than converting not to `HSBType`. We are much more consistent if we try to use as much `HSBType` as possible. If we want to make RGB a first-class citizen then we should discuss`RGBType`, but that is far out of scope here.

Reply:
;) not to mention XYType ..

Reply:
done, moved content of function to ```ColorUtilTest::assertRgbEquals``` (still needed for the test)